src/NHSD.BuyingCatalogue.Identity.Api/Controllers/AccountController.cs

```diff
@@ -99,6 +105,28 @@ public async Task<IActionResult> Login(LoginViewModel viewModel)
             throw new Exception("Invalid return URL");
         }
 
+        [HttpGet]
+        public async Task<IActionResult> Logout(string logoutId)
+        {
+            return await Logout(new LogoutViewModel
+            {
+                LogoutId = logoutId
+            });
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Logout(LogoutViewModel logoutViewModel)
+        {
+            logoutViewModel.ThrowIfNull(nameof(logoutViewModel));
```

Comment:
Is a null or empty `logoutViewModel.LogoutId` a valid scenario? As we'll end up with a null `logoutRequest` when `logoutViewModel.LogoutId` is null or empty and  `Redirect` will throw in that case is it worth adding `Required` attribute to the `LogoutViewModel.LogoutId`? We can then check whether `ModelState.IsValid`.

Reply:
I have removed this method & moved the logic into the GET logout method. Do you think the required tag would still apply?

Reply:
I wouldn't think so as there won't be any model binding happening. Is `LogoutViewModel` also still a view model?

Reply:
👍